### PR TITLE
[INLONG-2625][Sort] Support extracting data time in deserialization

### LINKED ISSUE #2625: [Feature][Sort] Support extracting data time in deserialization
### Description

Support extracting data time in deserialization

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by KevinWen007**
### Title Name: [INLONG-2625][Sort] Support extracting data time in deserialization

Fixes #2625 

### Motivation

Support extracting data time in deserialization

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### CODE REVIEW

File: inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/Entrance.java [hunk 1]
```diff
@@ -121,15 +125,22 @@ private static DataFlowInfo getDataflowInfoFromFile(String fileName) throws IOEx
...
                 .uid(Constants.DESERIALIZATION_SCHEMA_UID)
                 .name("Deserialization")
                 .setParallelism(config.getInteger(Constants.DESERIALIZATION_PARALLELISM));
     }
 
+    private static FieldInfo[] extractNonBuiltInFieldInfo(FieldInfo[] fieldInfos) {
+        return Arrays.stream(fieldInfos).filter(fieldInfo -> !(fieldInfo instanceof BuiltInFieldInfo)).toArray(
+                FieldInfo[]::new);
+    }
+
```
Comment (ywh19911024) [inline inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/Entrance.java:143]: 这个方法放到CommonUtil这个工具类中吧

File: inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/deserialization/FieldMappingTransformer.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.apache.flink.shaded.guava18.com.google.common.base.Preconditions.checkNotNull;
+
+public class FieldMappingTransformer {
```
Comment (ywh19911024) [inline inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/deserialization/FieldMappingTransformer.java:43]: 这个类可以增加些测试case

Comment (KevinWen007) [reply inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/deserialization/FieldMappingTransformer.java:43]: done

File: inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/Entrance.java (hunk 1 above)
Comment (KevinWen007) [reply inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/Entrance.java:143]: I don't think it's necessary to move it to CommonUtils currently.

**Review by gosonzhang — APPROVED** [review]

**Review by chantccc — APPROVED** [review]
+1

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**